codepy/toolchain.py

```diff
@@ -400,26 +393,23 @@ def strip_prefix(pfx, value):
 
     # on Mac OS X, "libraries" can also be "frameworks"
     libraries = []
-    for lib in make_vars["LIBS"].split():
+    for lib in config_vars["LIBS"].split():
         if lib.startswith("-l"):
-            libraries.append(strip_prefix("-l", lib))
+            libraries.append(lib[2:])
         else:
             cflags.append(lib)
 
+    ld, *ldflags = config_vars["LDSHARED"].split()
+    so_ext = os.path.splitext(config_vars["EXT_SUFFIX"])[-1]
```

Comment:
The previous version seemed to assume that `make_vars["SO"]` returned an extension, but at least on my system it doesn't exist at all. `config_vars["SO"]` exists though (from `sysconfig`), but that returns the `cpython-[arch].so`.

It also says that "SO" is deprecated in favor of "EXT_SUFFIX".

Reply:
Actually, it's `.cpython-[arch].so`.. was that the intended full suffix here?

Reply:
I think that's OK. As long as CPython looks for `xyz.cpython-[arch].so` for `import xyz`, that's all that's needed.